client/question/services/EventHandlerService.js

```diff
@@ -19,7 +19,38 @@
 tie.factory('EventHandlerService', [
   '$http', 'ServerHandlerService', function($http, ServerHandlerService) {
 
+
+    /**
+     * Global object to keep track of the current batch of events to send.
+     * @type [Event]
+     */
+    var currentEventBatch = [];
+
+    /**
+     * Submits the current batch of event data to TIE's backend.
+     *
+     */
+    var sendCurrentEventBatch = function() {
+      var data = {
+        events: currentEventBatch,
+        createdMsec: (new Date()).getTime()
+      };
+      return $http.post('/ajax/event/send_event_batch', data).then(
+        function() {
+          while (currentEventBatch.length > 0) {
```

Comment:
I think you can just do "currentEventBatch.length = 0" to make it empty again. (Weird, I know.) https://stackoverflow.com/questions/1232040/how-do-i-empty-an-array-in-javascript

Reply:
That is weird.